web/crux/src/app/template/template.dto.ts

```diff
@@ -0,0 +1,31 @@
+import { ApiProperty } from '@nestjs/swagger'
+import { IsIn, IsString, IsUUID } from 'class-validator'
+import { ProductTypeDto, PRODUCT_TYPE_VALUES } from '../product/product.dto'
+
+export class TemplateDto {
+  id: string
+
+  name: string
+
+  description?: string
+
+  technologies: string[]
+}
+
+export class CreateProductFromTemplateDto {
+  @IsUUID()
+  id: string
+
+  @IsString()
+  name: string
+
+  @IsString()
+  description?: string
+
+  @ApiProperty({
+    enum: PRODUCT_TYPE_VALUES,
+  })
```

Comment:
Needed?

Reply:
Yep. However the ` @IsString()` may not necessary.